spec/models/spree/payment_spec.rb

```diff
@@ -97,6 +97,30 @@
           expect { payment.process! }.to raise_error(Spree::Core::GatewayError)
           expect(payment.state).to eq('invalid')
         end
+
+        context "the payment is already authorized" do
+          before do
+            allow(payment).to receive(:response_code) { "pi_123" }
+          end
+
+          it "should call capture instead of purchase" do
+            expect(payment).to receive(:capture!)
+            expect(payment).to_not receive(:purchase!)
+            payment.process!
+          end
+        end
+      end
+
+      context "#process_offline when payment is already authorized" do
+        before do
+          allow(payment).to receive(:response_code) { "pi_123" }
+        end
+
+        it "should call capture if the payment is already authorized" do
+          expect(payment).to receive(:capture!)
+          expect(payment).to_not receive(:purchase!)
+          payment.process!
+        end
```

Comment:
I'm missing also an integration-level test that should ideally cover the exact case reported in #7495 , or as close as we can get.

Reply:
Okay, I added a spec to ..... the SubscriptionConfirmJob spec..... 😁 

It isn't perfect, but gets at the idea of what we want. Add it to the reasons to prioritize https://github.com/openfoodfoundation/openfoodnetwork/pull/7281 - and just to rework subscriptions in general.